### PR TITLE
Fix deleting referenced chains before end of graceful restart window.

### LINKED ISSUE #834: Error logs hit when deleting VM (after stop/starting felix)
Repro scenario.  On Openstack Icehouse with calico:
- create a VM on a host.
- stop felix on that host
- start felix on that host
- delete the VM.

At the time of deletion, we get 8 errors in felix.log like this:
2015-09-30 15:40:06,016 [ERROR][31781/8] calico.felix.fiptables 569: Failed to delete chain felix-p-_a15fd76a273ef0d-o, giving up. Maybe it  is still referenced?

Readily reproducible at present on Master.

### PR DESCRIPTION
**Pull request by fasaxc**
Prior to this fix, the chains would be left in place until the
end of the graceful restart window rather than being stubbed out
immediately.

Fixes #834.

### CODE REVIEW

File: calico/felix/fiptables.py [hunk 1]
```diff
@@ -612,30 +612,40 @@ def _calculate_ipt_modify_input(self):
...
         input_lines = []
-        affected_chains = self._txn.affected_chains
-        for chain in affected_chains:
-            if (self._grace_period_finished or
-                    chain not in self._chains_in_dataplane or
-                    chain not in self._txn.chains_to_stub_out):
-                # We're going to rewrite or delete this chain below, mark it
-                # for creation/flush.
-                input_lines.append(":%s -" % chain)
+        chains_to_flush = set()
```
Comment (matthewdupre) [inline calico/felix/fiptables.py]: Can you rename this to chains_to_stub.  Flush is a bit confusing - is it different?  (If I'm wrong and it doesn't mean stub then please talk to me!)

**Comment by matthewdupre** [comment]
Otherwise LGTM

**Comment by matthewdupre** [comment]
test this please

**Comment by fasaxc** [comment]
retest this please

File: calico/felix/fiptables.py (hunk 1 above)
Comment (fasaxc) [inline calico/felix/fiptables.py]: Changed to `modified_chains` and added a comment.

**Comment by fasaxc** [comment]
retest this please

**Pull request merged**